File: src/main/java/net/gtaun/shoebill/common/command/CommandParameter.java
package net.gtaun.shoebill.common.command;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by marvin on 15.02.16.
 * Copyright (c) 2015 Marvin Haschker. All rights reserved.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PARAMETER})
public @interface CommandParameter {
    String name() default "";
    String description() default "";
}

File: src/main/java/net/gtaun/shoebill/common/command/CommandParameter.kt
package net.gtaun.shoebill.common.command

/**
 * Created by marvin on 15.02.16.
 * Copyright (c) 2015 Marvin Haschker. All rights reserved.
 */
@Retention
@Target(AnnotationTarget.VALUE_PARAMETER)
annotation class CommandParameter(val name: String, val description: String = "")
